tests/main/snap-user-service-restart-on-upgrade/task.yaml

```diff
@@ -25,6 +25,7 @@ prepare: |
 
 restore: |
     tests.session -u test restore
+    snap remove test-snapd-user-service
```

Comment:
could we instead use tests.cleanup defer for these? I think it's clearer, but also if you want to do this as a follow-up that's fine too

Reply:
I'll do it in a follow up, still want to see if it really fix the issue